src/userbase-js/types/index.d.ts

```diff
@@ -0,0 +1,54 @@
+export as namespace Userbase;
+
+/* auth */
+
+export interface Session {
+  sessionId: string
+  creationDate: string
+}
+
+export interface UserProfile {
+  [key: string]: string
+}
+
+export interface UserResult {
+  username: string
+  key: string
+  email?: string
+  profile?: UserProfile
+}
+
+export function init(options: { appId: string, endpoint?: string, keyNotFoundHandler?: () => void }): Promise<Session>
+export function signUp(username: string, password: string, email: string | null, profile: UserProfile | null, showKeyHandler: () => void | null, rememberMe: boolean): Promise<UserResult>
```

Comment:
If they return a Promise does Userbase look at or use the resolution value?

Reply:
Nope, just waits for it to resolve ([source](https://github.com/encrypted-dev/userbase/blob/ff912715ac3a4642d6d3008ea42b256af6faf6d0/src/userbase-js/auth.js#L269)). This is like this because the default handler waits until the user [closes the modal](https://github.com/encrypted-dev/userbase/blob/ff912715ac3a4642d6d3008ea42b256af6faf6d0/src/userbase-js/auth.js#L240-L243), thus explicitly acknowledging they have seen and saved the key